contracts/teller-network/SellerStore.sol

```diff
@@ -0,0 +1,173 @@
+
+    enum PaymenMethods {Cash,BankTransfer,InternationalWire}
+    enum MarketType {Above, Below}
+
+    event Added(
+        address owner,
+        uint256 offerId,
+        address asset,
+        address statusContractCode,
+        string location,
+        string currency,
+        string username,
+        PaymenMethods[] paymentMethods,
+        MarketType marketType,
+        uint8 margin
+    );
+
+    event Updated(
+        address owner,
+        uint256 id,
+        address asset,
+        address statusContractCode,
+        string location,
+        string currency,
+        string username,
+        PaymenMethods[] paymentMethods,
+        MarketType marketType,
+        uint8 margin
+    );
+
+    struct Seller {
+        address statusContractCode;
+        string location;
+        string username;
+    }
+
+    struct Offer {
+        address asset;
+        string currency;
+        uint8 margin;
+        PaymenMethods[] paymentMethods;
+        MarketType marketType;
+    }
+
+    address public license;
+    Seller[] public sellers;
+    Offer[] public offers;
+    mapping(address => bool) public sellerWhitelist;
+    mapping(address => uint256) public addressToSeller;
+
+    mapping(address => mapping (uint256 => bool)) public offerWhitelist;
+    mapping(address => uint256[]) public addressToOffers;
+
+    constructor(address _license) public {
+        license = _license;
+    }
+
+    function setLicense(address _license) public onlyOwner {
+        license = _license;
+    }
+
+    /**
+    * @dev Add a new offer with a new seller if needed to the list
+    * @param _asset The address of the erc20 to exchange, pass 0x0 for Eth
+    * @param _statusContractCode The address of the status contract
+    * @param _location The location on earth
+    * @param _currency The currency the seller want to receive (USD, EUR...)
+    * @param _username The username of the seller
+    * @param _paymentMethods The list of the payment methods the seller accept
+    * @param _marketType Above or Below
+    * @param _margin The margin for the seller from 0 to 100
+    */
+    function add(
+        address _asset,
+        address _statusContractCode,
+        string memory _location,
+        string memory _currency,
+        string memory _username,
+        PaymenMethods[] memory _paymentMethods,
+        MarketType _marketType,
+        uint8 _margin
+    ) public {
+        require(License(license).isLicenseOwner(msg.sender), "Not a license owner");
+        require(_margin <= 100, "Margin too high");
+
+        if (!sellerWhitelist[msg.sender]) {
+            Seller memory seller = Seller(_statusContractCode, _location, _username);
+            uint256 sellerId = sellers.push(seller) - 1;
+            addressToSeller[msg.sender] = sellerId;
+            sellerWhitelist[msg.sender] = true;
+        } else {
+            sellers[addressToSeller[msg.sender]].statusContractCode = _statusContractCode;
```

Comment:
if we assign sellers[addressToSeller[msg.sender]] to a temporary variable like  `Seller storage tmpSeller`, and then refer to that tmp variable, does the cost increases / decreases?

Reply:
The cost decrease slightly